### DISCUSSION TITLE
bug: WalletConnect connector sending network requests as soon as it's created, and autoConnect initializes all connectors, not just the last one

### DISCUSSION THREAD

**Discussion by wille** [wevm/wagmi#3085] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

All of them

### Current Behavior

As soon as the WalletConnect connector is initialized, which commonly happens in the entry point of your web application, the full underlying library will load and initialize, opening websockets and fetching images.

Same situation for the coinbase connector, but only if autoConnect is set.

Loading these extra libs unless necessary slows down websites and I do not want to send outgoing requests to these domains unless the user has explicitly used any of these connectors.

![image](https://github.com/wagmi-dev/wagmi/assets/1017440/fba818f4-83bd-473a-8755-2d8125c4823f)
![image](https://github.com/wagmi-dev/wagmi/assets/1017440/12cc0303-5905-473f-9a57-9e66eab93c69)

### Expected Behavior

For WalletConnect, we can start with not initializing the library in the constructor, so it only happens when using it or if autoConnect is set to true.
Also if `showQrModal` is false in the initial options passed to the WC connector, then it will avoid loading the images in the background , but the QR code will never show up. I think it's worth exposing a function to trigger the QR modal, even if the connector is already initialized.

For Coinbase, the WalletLink RPC call is only sent if autoConnect is true, so I'm thinking that autoConnect should only connect to the latest known used connector by the client and not initialize all of them in ascending order if none is authorized.

### Steps To Reproduce

This can be seen live at https://wagmi.sh/

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

https://wagmi.sh/

### Anything else?

Looking for your feedback and thoughts. I will create a PR and run the changes on bitrefill.com

**Comment by tmm** [comment]
Thanks for the thoughts. Moved to discussion.

> For WalletConnect, we can start with not initializing the library in the constructor.

Probably won't change this behavior for Wagmi v1, but could consider it for Wagmi v2. Feel free to take a look at the new connector API on the `alpha` branch and comment on #3068 

>  I think it's worth exposing a function to trigger the QR modal

What would this API look like? Ideally all connectors follow the same API.

> autoConnect should only connect to the latest known used connector

You can have multiple connections in Wagmi v2 so we want to reconnect to all of them. Also, if someone doesn't use storage, we won't know if any have been previously connected.

**Comment by wille** [comment]
OK so there is a wagmi v2? I spent lots of time upgrading from 0.x to 1.x to hopefully have this fixed. I'll have to run a fork for now then.

> What would this API look like? Ideally all connectors follow the same API.

Disregard this one, just realized I can just use getProvider().modal.openModal()

> You can have multiple connections in Wagmi v2 so we want to reconnect to all of them. Also, if someone doesn't use storage, we won't know if any have been previously connected.

Sure, but if there is storage and you are aware of connectors, only connect to the used ones, and not all of them

**Comment by wille** [comment]
> Probably won't change this behavior for Wagmi v1, but could consider it for Wagmi v2. Feel free to take a look at the new connector API on the alpha branch and comment on https://github.com/wagmi-dev/wagmi/discussions/3068

Cannot see why not, it will not change anything, except that it loads later, only when needed
